README.md

```diff
@@ -1,12 +1,33 @@
 # plearn
+What is this?
+
+Explanation 1:
+
+  Duolingo for programming languages.
+
+Explanation 2 (Algorithmically Selected "Chapter Exercises"):
+
+  You're studying a textbook (do those still exist?).  At the end of the
+  chapter are problems that help you to:
+
+    * practice knowledge skills
+    * learn what you don't know
+    * learn what you do know
+
+ This tool uses information about you (bound to a pseudonym), primarily the 
+ history of your exercise responses, and others who are interested in the same
+ topic to guess which _next_ exercise will teach you the most.
```

Comment:
This informs what the model will consist of.  What objects are we binding to the user?  A history of responses, sure, but also we need to extract essential information from the response (rather than computing it upon each guess of next exercise).  I'm thinking scoring in several categories:

* efficiency
* elegance (wtf is that?)
* abstractness (sort of like "flexibility"?)

These are just ideas, but certainly something that will be required for the model.  Anyway, this is off topic, so I'll just end it there.